src/rmt.rs

```diff
@@ -856,3 +915,189 @@ mod chip {
+        }
+
+        Ok(Self {
+            channel: C::channel() as _,
+            _p: PhantomData,
+            pulse_pair_vec,
+        })
+    }
+
+    pub fn channel(&self) -> rmt_channel_t {
+        self.channel as _
+    }
+
+    /// Start receiving
+    pub fn start(&self) -> Result<(), EspError> {
+        esp!(unsafe { rmt_rx_start(self.channel(), true) })
+    }
+
+    /// Stop receiving
+    pub fn stop(&self) -> Result<(), EspError> {
+        esp!(unsafe { rmt_rx_stop(self.channel()) })
+    }
+
+    // Set ticks_to_wait to 0 for non-blocking.
+    pub fn get_rmt_items(&mut self, ticks_to_wait: u32) -> Result<u32, EspError> {
+        //let mut rmt_handle: RingbufHandle_t = std::ptr::null_mut();
+        let mut rmt_handle: RingbufHandle_t = ptr::null_mut();
+        let mut length: u32 = 0;
+
+        unsafe {
+            esp!(rmt_get_ringbuf_handle(self.channel(), &mut rmt_handle))
+                .expect("Failed to get ringbuffer handle");
+
+            let rmt_items = xRingbufferReceive(rmt_handle as *mut _, &mut length, ticks_to_wait)
+                as *mut rmt_item32_t;
+
+            if length > 0 {
+                let rmt_item_qty: u32 = length / 4;
+
+                self.pulse_pair_vec.clear();
+
+                for n in 0..rmt_item_qty {
+                    let lvl0: u32 = (*rmt_items.offset(n.try_into().unwrap()))
+                        .__bindgen_anon_1
+                        .__bindgen_anon_1
+                        .level0();
+                    let dur0: u32 = (*rmt_items.offset(n.try_into().unwrap()))
+                        .__bindgen_anon_1
+                        .__bindgen_anon_1
+                        .duration0();
+                    let lvl1: u32 = (*rmt_items.offset(n.try_into().unwrap()))
+                        .__bindgen_anon_1
+                        .__bindgen_anon_1
+                        .level1();
+                    let dur1: u32 = (*rmt_items.offset(n.try_into().unwrap()))
+                        .__bindgen_anon_1
+                        .__bindgen_anon_1
+                        .duration1();
+
+                    self.pulse_pair_vec
+                        .push(PulsePair::new(lvl0, dur0, lvl1, dur1));
+                }
+
+                vRingbufferReturnItem(rmt_handle, rmt_items as *mut c_types::c_void);
+            }
+        }
+
+        Ok(length)
+    }
+}
+
+impl<'d> Drop for RxRmtDriver<'d> {
+    /// Stop receiving and release the driver.
+    fn drop(&mut self) {
+        self.stop().unwrap();
+        esp!(unsafe { rmt_driver_uninstall(self.channel()) }).unwrap();
+    }
+}
+
+unsafe impl<'d> Send for RxRmtDriver<'d> {}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct PulsePair {
+    pub level0: PinState,
+    pub duration0: u16,
+    pub level1: PinState,
+    pub duration1: u16,
+}
+
+impl PulsePair {
+    pub fn new(lvl0: u32, dur0: u32, lvl1: u32, dur1: u32) -> Self {
```

Comment:
Given that this constructor takes "raw" ESP IDF values of type `u32` I don't think it should be public.